qiskit_ibm_runtime/noise_learner.py

```
@@ -0,0 +1,276 @@
+                * A :class:`Session` if you are using session execution mode.
+                * A :class:`Batch` if you are using batch execution mode.
+
+                Refer to the `Qiskit Runtime documentation <https://docs.quantum.ibm.com/run>`_.
+                for more information about the ``Execution modes``.
+
+            backend: Backend to run the primitive. This can be a backend name or an :class:`IBMBackend`
+                instance. If a name is specified, the default account (e.g. ``QiskitRuntimeService()``)
+                is used.
+
+            session: Session in which to call the primitive.
+
+                If both ``session`` and ``backend`` are specified, ``session`` takes precedence.
+                If neither is specified, and the primitive is created inside a
+                :class:`qiskit_ibm_runtime.Session` context manager, then the session is used.
+                Otherwise if IBM Cloud channel is used, a default backend is selected.
+
+            options: :class:`NoiseLearnerOptions`. Alternatively, :class:`EstimatorOptions` can be
+                provided for convenience, in which case the estimator options get reformatted into
+                noise learner options and all the irrelevant fields are ignored.
+        """
+        self._mode: Optional[Union[Session, Batch]] = None
+        self._service: QiskitRuntimeService | QiskitRuntimeLocalService = None
+        self._backend: Optional[BackendV1 | BackendV2] = None
+
+        if isinstance(mode, (Session, Batch)):
+            self._mode = mode
+            self._service = self._mode.service
+            self._backend = self._mode._backend
+        elif isinstance(mode, IBMBackend):  # type: ignore[unreachable]
+            self._service = mode.service
+            self._backend = mode
+        elif isinstance(mode, (BackendV1, BackendV2)):
+            self._service = QiskitRuntimeLocalService()
+            self._backend = mode
```

This block is for local testing mode (i.e. if an Aer simulator or a FakeBackend is passed in). It should raise an exception because NoiseLearner is not supported in local mode (yet).